scripts/zones/Temple_of_Uggalepih/npcs/qm15.lua

```diff
@@ -17,7 +17,7 @@ entity.onTrigger = function(player, npc)
         player:getCharVar('KnightStalker_Progress') == 4 and
         player:getCharVar('KnightStalker_Kill') == 0 and
         player:getMainJob() == xi.job.DRG and
-        pet and
+        pet ~= nil and
```

Comment:
`player:getPet()` can only return a valid CLuaBaseEntity or nil, checking `pet` and `pet ~= nil` are equivalent if the pet has come out correctly, what did you observe here?

Reply:
```cpp

std::optional<CLuaBaseEntity> CLuaBaseEntity::getPet()
{
    auto* PBattle = static_cast<CBattleEntity*>(m_PBaseEntity);

    if (PBattle->PPet != nullptr)
    {
        return std::optional<CLuaBaseEntity>(PBattle->PPet);
    }

    return std::nullopt;
}
```

Reply:
Might end up closing this. Could be something weird on my end? I wasn't getting any lua errors. When I restarted my server/client, it happened to work fine afterwards. Before it was just not popping or displaying a message.